### DISCUSSION TITLE
Auto-save function ?

### DISCUSSION THREAD

**Discussion by vi-tan** [stakira/OpenUtau#498] [Ideas]
If there's anything I was incredibly thankful for when I used UTAU-synth was the fact you could recover what you were working on if the app crashed .
I was workin on a ust , two hours in , and I didn't realize the app was updating ( I pressed X on the update screen cause it was takin too long ) and when it was done , it booted me out . Fear ran through me at the thought of my progress being reduced to nothing and - it did . Not sure if Auto-save is possible , especially if a situation like mine happens again but just a thought .

**Comment by GenroidKwame-Dev** [comment]
What OS are you using coz on Windows I know OU has AutoSave function that saves your ust every few minutes and names it as "filename"-autosave.ustx. I assume that only works if you have saved it as ustx once. There is a recovery option in the PR if I'm not wrong so it might be implemented soon

**Reply by vi-tan** [reply]
I use mac , and ah I see . That makes sense , guess I'll be careful to do that next time . Thank you !
